File: testdata/target_complex.ts
// target_complex.ts

export function simpleTsFunc(): void {
    console.log("Old simpleTsFunc from target_complex.ts");
}

export const arrowTsFunc = (param1: string, param2: number): boolean => {
    console.log(`Old arrowTsFunc from target_complex.ts: ${param1}, ${param2}`);
    return false;
};

class MyTsClass {
    constructor(private id: number) {}

    public classMethod(data: any): string {
        console.log(`Old MyTsClass.classMethod from target_complex.ts, id: ${this.id}, data:`, data);
        return "old_class_method_result";
    }

    static staticTsMethod(): void {
        console.log("Old MyTsClass.staticTsMethod from target_complex.ts");
    }

    public keepThisMethod(): void {
        console.log("MyTsClass.keepThisMethod from target_complex.ts - I should remain.");
    }
}

async function asyncTsFunc(name: string): Promise<string> {
    console.log(`Old asyncTsFunc from target_complex.ts with ${name}`);
    await new Promise(resolve => setTimeout(resolve, 5));
    return `Async hello ${name} from target`;
}

function utilityTsFunc() {
    console.log("Old utilityTsFunc from target_complex.ts");
}

export function targetSpecificTsFunc(): void {
    console.log("targetSpecificTsFunc in target_complex.ts");
}

// A commented out version, should not be touched
// export function simpleTsFunc(): void {
//    console.log("Commented out simpleTsFunc");
// }

export function genericTsFunc<T>(arg: T): T {
    console.log("Old genericTsFunc from target_complex.ts", arg);
    return arg;
}

// Some code after all functions
export const endMarkerTargetComplexTs = true;

File: testdata/target_complex.go
package main

import "fmt"

// Old version of SimpleFunc
func SimpleFunc() {
	fmt.Println("Old SimpleFunc from target_complex.go")
}

type Receiver struct {
	Value int
}

// Old version of MethodWithArgs
func (r *Receiver) MethodWithArgs(a int, b string) (bool, error) {
	fmt.Println("Old MethodWithArgs from target_complex.go", r.Value, a, b)
	return false, fmt.Errorf("old error")
}

func KeepThisFunc() {
	fmt.Println("KeepThisFunc from target_complex.go - I should remain.")
}

// This function is not in source_complex.go, should remain.
func TargetSpecificFunc() {
	fmt.Println("TargetSpecificFunc in target_complex.go")
}

// This method exists in source_complex.go, so it should be replaced.
func (s *Service) Serve() {
    fmt.Println("Old Service.Serve from target_complex.go")
}

type Service struct{}

func FuncWithNoReceiver() string { // This one should be replaced
	return "Old FuncWithNoReceiver from target_complex.go"
}

/*
func (r *Receiver) MethodWithArgs(a int, b string) (bool, error) {
	// A commented out version, should not be touched
	fmt.Println("Commented out MethodWithArgs")
	return true, nil
}
*/

// Some code after all functions to test where new functions might be added
var EndMarkerTargetComplexGo = true
